src/main/java/org/opentripplanner/routing/edgetype/BikeParkEdge.java

```diff
@@ -49,16 +49,19 @@ protected State traverseUnpark(State s0) {
         /*
          * To unpark a bike, we need to be walking, and be allowed to bike.
          */
-        if (s0.getNonTransitMode() != TraverseMode.WALK || !options.modes.getBicycle())
-            return null;
```

Comment:
I know it's a common pattern in the imperative Java world but I find "return early" awful for code readability.

Here are my reasons:

- an `if` should always have an `else` to avoid unintended fall throughs
- in this instance it relies on negated boolean expressions which add semantic weight, asking the reader to think about what the conditions _are not_ (many more possibilities), rather than on what they _are_
- it draws attention to the fall-through case, rather than the actual business logic

However, in the interest of not starting a pointless style war, I will relent and reverse the order, if you feel strongly about it.